mail_tracking_mailgun/models/mail_tracking_email.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import hashlib
+import hmac
+from datetime import datetime
+from openerp import models, api, fields
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class MailTrackingEmail(models.Model):
+    _inherit = "mail.tracking.email"
+
+    def _country_search(self, country_code):
+        country = False
+        if country_code:
+            country = self.env['res.country'].search([
+                ('code', '=', country_code.upper()),
+            ])
+        if country:
+            return country.id
+        return False
+
+    @property
+    def _mailgun_mandatory_fields(self):
+        return ('event', 'timestamp', 'token', 'signature',
+                'tracking_email_id', 'odoo_db')
+
+    @property
+    def _mailgun_event_type_mapping(self):
+        return {
+            # Mailgun event type: tracking event type
+            'delivered': 'delivered',
+            'opened': 'open',
+            'clicked': 'click',
+            'unsubscribed': 'unsub',
+            'complained': 'spam',
+            'bounced': 'hard_bounce',
+            'dropped': 'reject',
+        }
+
+    @property
+    def _mailgun_supported_event_types(self):
+        return self._mailgun_event_type_mapping.keys()
+
+    def _mailgun_event_type_verify(self, event):
+        event = event or {}
+        mailgun_event_type = event.get('event')
+        if mailgun_event_type not in self._mailgun_supported_event_types:
```

Comment:
I think you can drop `_mailgun_supported_event_types` and just use `_mailgun_event_type_mapping`:

```
>>> "a" in {"a": 1, "b": 2}
True
```